rpc/src/rpc/native/trie.rs

```diff
@@ -0,0 +1,169 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use ethereum_types::H256;
+use ethers::{
+    types::Bytes,
+    utils::{keccak256, rlp},
+};
+use mpt_trie::{
+    nibbles::Nibbles,
+    partial_trie::{Node, PartialTrie, WrappedNode},
+};
+
+/// The hash of an empty trie.
+pub const EMPTY_TRIE_HASH: H256 = H256([
+    0x56, 0xe8, 0x1f, 0x17, 0x1b, 0xcc, 0x55, 0xa6, 0xff, 0x83, 0x45, 0xe6, 0x92, 0xc0, 0xf8, 0x6e,
+    0x5b, 0x48, 0xe0, 0x1b, 0x99, 0x6c, 0xad, 0xc0, 0x01, 0x62, 0x2f, 0xb5, 0xe3, 0x63, 0xb4, 0x21,
+]);
```

Comment:
non-blocking, as not exposed (yet) but we should try to group all these common constants somewhere (probably somewhere under `zk_evm` crates) and reuse/re-exports them instead of always redefining them

Reply:
Agreed, I will add a TODO. I was also wondering if this module would be better positioned in the `zk_evm/mpt_trie` crate for reuse?

Reply:
yeah that'd make sense, it's a bit too specific on MPTs to be inside `zero-bin` I'd say.
cc @0xaatif @BGluth

Reply:
Yeah I think inside `mpt_trie` probably makes the most sense.